aten/src/ATen/native/nested/README.md

```diff
@@ -0,0 +1,63 @@
+
+##  Implementing new functions
+There are two main classes of functions that can be implemented on NestedTensors: functions that can be efficiently implemented by viewing the NestedTensor as dense tensor where the raggedness has been folded into the dense dimensions and those that can't. Unary operations (e.g. abs, log, relu) and binary operations (e.g. add, mul, div) that act elementwise on one or more NestedTensors are examples of functions that can be efficiently implemented. Efficient implementation of these functions is relatively straightforward.
+
+The definition of map_nt is:
+
+```cpp
+template <typename Func>
+Tensor map_nt(const Tensor& nt, Func f) {
+  auto* nt_impl = get_nested_tensor_impl(nt);
+  const auto& sizes = nt_impl->get_nested_size_tensor();
+  return at::detail::make_tensor<NestedTensorImpl>(f(nt_impl->get_buffer()), sizes);
+}
+```
+1. Get the NestedTensorImpl from the input NestedTensor.
+2. Get the sizes of the NestedTensor.
+3. Call get_buffer() which returns a flat, dense tensor whose storage shares that of the input NestedTensor.
+4. Call the function f on the dense tensor.
+5. Construct a new NestedTensor from the output of f and the sizes of the input NestedTensor.
+
+There are also important functions that, under certain conditions of regularity, can be implemented effectively by accessing the underlying buffer and viewing it in a special manner. For a good example of this see the implementation of `linear` in [NestedTensorTransformerFunctions.cpp](NestedTensorTransformerFunctions.cpp).
+
+The second class of functions can't be efficiently implemented by viewing the NestedTensor as a dense tensor. An example of this is `softmax_nested`. The implementation of this function can be found in [NestedTensorMath.cpp](NestedTensorMath.cpp). This can occur when there are strict problem boundaries that occur at the tensor_component level. For example, when computing the softmax of a NestedTensor we need to compute the softmax of each tensor_component individually. This is because the softmax function is not defined on tensors of different shapes. In this case we can't view the NestedTensor as a dense tensor and apply the softmax function to the entire tensor at once. Instead we need to apply the softmax function to each tensor_component individually.
+
+The problem with this though is that iterating over a potentially large number of tensor_components and launching individual cuda kernels for each one is very inefficient. Ideally we would launch one kernel that operates on all the tensor_components in parallel.
+
+If performance is not your main concern and you would like to enable coverage the function `map_nested_tensor` can be found in [NestedTensorUtils.h](NestedTensorUtils.h). This function iterates over tensor components to applying a function to each tensor_component individually. Be warned though that this is likely very inefficient, especially for cuda implementations of the function.
```

Comment:
nit: For CPU implementations this can actually serve as a pretty good baseline. If we had nested parallelism down we could offer a `parallel_map_nested_tensor`.

Reply:
add a couple openmp parellel for and bobs your uncle. I remember Yifan was trying to get this to work at one point but was not succesful. I will note that though